### PR TITLE
Fix Dockerfile URL support in build configuration

### LINKED ISSUE #4464: flyctl v0.3.149 misinterprets Dockerfile URL in `[build]` as a local path
---

**Describe the bug**
When a URL is supplied for `[build] dockerfile` in `fly.toml`, `flyctl` concatenates the URL to the local working-directory path instead of downloading the file. It then fails with “Dockerfile … not found”.

* Operating system: Ubuntu 22.04 LTS
* `fly version`: v0.3.149

---

**Paste your `fly.toml`**

```toml
app = "shmuel-tech-pg-formula"
primary_region = "ams"
kill_signal = "SIGINT"

[build]
dockerfile = "https://raw.githubusercontent.com/skamensky/postgres-formula-compiler/master/Dockerfile"

[env]
  SERVICE_NAME = "pg-formula"
  PORT = "80"

[http_service]
  internal_port = 80
  force_https = true
  auto_stop_machines = "stop"
  auto_start_machines = true
  min_machines_running = 0

  [[http_service.checks]]
    path = "/health"
    method = "GET"
    interval = "30s"
    timeout = "5s"
    grace_period = "10s"
```

---

**Command output:**

```text
shmuel@TopLap123:~/repos/shmuel.tech/services/pg-formula$ fly deploy --remote-only --verbose
==> Verifying app config
Validating /home/shmuel/repos/shmuel.tech/services/pg-formula/fly.toml
✓ Configuration is valid
--> Verified app config
==> Building image
==> Building image
Error: failed to fetch an image or build from source: dockerfile '/home/shmuel/repos/shmuel.tech/services/pg-formula/https:/raw.githubusercontent.com/skamensky/postgres-formula-compiler/master/Dockerfile' not found
```

---

### Expected behavior

`flyctl` should recognise the `https://` scheme, download the Dockerfile into the temporary build directory, and hand the correct path to BuildKit.

### Additional context

Docs state that `dockerfile` “accepts … a URL”, but the current stable CLI (v0.3.149) treats the value as a relative filesystem path. [Docs claiming URL support](https://web.archive.org/web/20250628185145/https://fly.io/docs/reference/configuration/#specify-a-dockerfile)

A similar path-handling issue exists for the `--dockerfile` flag (#400).

### PR DESCRIPTION
**Pull request by rubys**
## Summary
- Fix issue where flyctl misinterprets Dockerfile URLs in `[build]` section as local paths
- Add URL detection and download functionality for Dockerfile URLs
- Implement comprehensive error handling for HTTP requests

## Changes Made
- **Modified `resolveDockerfilePath()`** to detect and handle HTTP/HTTPS URLs
- **Added `downloadDockerfileFromURL()`** to download Dockerfiles from URLs to temporary files
- **Added `isURL()`** helper function to validate URL format
- **Enhanced test coverage** with comprehensive test cases for URL and local path scenarios

## Test Coverage
- ✅ Local Dockerfile path resolution (existing functionality)
- ✅ HTTP/HTTPS URL Dockerfile download
- ✅ Error handling for 404 responses
- ✅ URL validation edge cases
- ✅ All existing tests continue to pass

## Fixes
Closes #4464

## Test Plan
- [x] Run existing tests to ensure no regressions
- [x] Test URL-based Dockerfile download functionality
- [x] Test error handling for invalid URLs and HTTP errors
- [x] Verify lint and formatting compliance

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: internal/command/deploy/deploy_build.go [hunk 1]
```diff
@@ -234,17 +239,73 @@ func determineImage(ctx context.Context, appConfig *appconfig.Config, useWG, rec
...
+	tempFile, err := os.CreateTemp("", "dockerfile-*.tmp")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temporary file: %w", err)
+	}
+	defer tempFile.Close()
+
+	// Download the Dockerfile
+	resp, err := http.Get(dockerfileURL)
+	if err != nil {
+		os.Remove(tempFile.Name())
```
Comment (kzys) [inline internal/command/deploy/deploy_build.go:254]: Can you move os.Remove() to defer? You can check err in defer to see whether the function is successfully finished or not.
